rustual-boy-core/src/rom.rs

```diff
@@ -22,16 +22,22 @@ impl Rom {
         let mut vec = Vec::new();
         file.read_to_end(&mut vec)?;
 
-        let size = vec.len();
+        Rom::from_bytes(&vec)
+    }
+
+    pub fn from_bytes(bytes: &[u8]) -> io::Result<Rom> {
```

Comment:
Hehe, yeah, I can see why that's desirable, but here I'd advocate the extra generality/convenience is more important than avoiding a copy that happens once at startup, and is only a few megs max.